### DISCUSSION TITLE
Performance problem

### DISCUSSION THREAD

**Discussion by nikolavojicic** [clj-kondo/clj-kondo#1400] [Q&A] [answered]
Hi,

I'm using `clj-kondo` via `flycheck-clj-kondo` on two similarly powerful machines.  
On one I get nice performance, on the other one it's slow, almost unusable.  
Both are using the same versions:

- `Windows 10`
- `GNU Emacs 27.1 (build 1, x86_64-w64-mingw32) of 2020-08-21`
- `clj-kondo-2021.09.25-windows-amd64`
- `flycheck-31`
- `flycheck-clj-kondo-0.0.3`
- `clojure 1.10.3`
- `jdk 16.0.1.9.1`

Running `clj-kondo --lint ...` from the PowerShell has the same performance on both PCs.  
Any advice how to track down the problem?

Thanks.

**Comment by borkdude** [comment]
@nikolavojicic If you use `flycheck-compile` then you will see the command that flycheck runs to invoke clj-kondo. Perhaps that helps to debug?

**Reply by nikolavojicic** [reply]
It prints the same output, on both:
```
-*- mode: compilation; default-directory: "c:/..." -*-
Compilation started at ...

"clj-kondo" "--lint" "-" "--lang" "clj" "--filename" "c:/.../api.clj" "--cache" < "c:/.../api.clj"
linting took 28ms, errors: 0, warnings: 0

Compilation finished at ...
```
BTW on the one where overall performance is fine, `--lint` runs slower, cca 38ms.

**Comment by nikolavojicic** [comment, marked as answer]
I saw `Antimalware Service Executable` burning in the `Task Manager` as I type.
So the solution is to add `clj-kondo` folder / exe to `Windows Defender` -> `Exclusions`.
On the other PC I have ESET instead, and `ESET Service (2)` also takes some CPU as I type, but it's faster than `Windows Defender`.
I'll exclude `clj-kondo` there as well.
Thanks!

**Reply by borkdude** [reply]
Thank you!
